src/goto-instrument/contracts/assigns.cpp

```diff
@@ -192,14 +202,26 @@ void assigns_clauset::remove_from_write_set(const exprt &target_expr)
 }
 
 exprt assigns_clauset::generate_inclusion_check(
-  const conditional_address_ranget &lhs) const
+  const conditional_address_ranget &lhs,
+  optionalt<cfg_infot> &cfg_info_opt) const
 {
   if(write_set.empty())
     return not_exprt{lhs.validity_condition_var};
 
   exprt::operandst conditions{not_exprt{lhs.validity_condition_var}};
-  for(const auto &target : write_set)
-    conditions.push_back(target.generate_unsafe_inclusion_check(lhs));
+
+  if(cfg_info_opt.has_value())
+  {
+    auto &cfg_info = cfg_info_opt.value();
+    for(const auto &target : write_set)
+      if(target.maybe_alive(cfg_info))
```

Comment:
```suggestion
    for(const auto &target : write_set)
      if(target.maybe_alive(cfg_info_opt.value()))
```

Reply:
this would reintroduce n function calls to `optional::value()`